### ISSUE TITLE
🐛 No output for chmod file mode change

### ISSUE THREAD

**Issue by mrjoel** [dandavison/delta#583]
There is no output shown for file mode change (add/remove execute bit). I've tested using delta 0.7.1, with my custom style configuration as well as the default style (empty delta section in effective `.gitconfig`). Examples of raw git output is shown below, for both for both a mode only change, as well as mode changes and content changes

I don't offhand have any thrilling recommendation as to an alternate display (compact single line `+x` or `-x`?), but even just allowing the plaintext through delta would be very beneficial. Otherwise what can be a noteworthy change is hidden entirely.

```
diff --git i/foo w/foo
old mode 100644
new mode 100755
```

```
diff --git i/foo w/foo
old mode 100644
new mode 100755
index 25160a3..57d50cf
--- i/foo
+++ w/foo
@@ -1 +1,2 @@
 line1
+line2
```

**Comment by dandavison** [comment]
Thanks @mrjoel , @HarHarLinks -- agree this needs fixing, and it should be an easy enough fix.

**Comment by dandavison** [comment]
I've merged a fix. I initially went for this:

```
src/delta.rs: 100644 ⟶   100755
```

but I've now changed that to this (reasoning below)

```
src/delta.rs: mode +x
```

-------------------------------------------------

Firstly, I do see that there's an argument for having the old and new modes vertically aligned in order to compare bits, and futhermore

> similar to ls -l (or colored like exa -l ), and finally I expect people would like only a diff variant similar to chmod u+x syntax

@HarHarLinks I was all set to implement this.

However, I realized that git does not actually output all file mode permissions changes into the diff. I'm not sure if it's in the official docs but [this blog post](https://medium.com/@tahteche/how-git-treats-changes-in-file-permissions-f71874ca239d) gives a good explanation, from which

> 100755 for executable and 100644 for non-executable are the only file modes Git records.

So, basically, I think perhaps what really makes sense is for delta to simply emit either `mode -x` or `mode +x`, as the [git diff --compact-summary](https://git-scm.com/docs/git-diff-files#Documentation/git-diff-files.txt---compact-summary) does: 

```
$ git diff --compact-summary | cat
 src/delta.rs (mode +x) | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
```

**Comment by HarHarLinks** [comment]
Interesting, TIL. Considering this, I agree with your decision. Thank you for tackling this.

**Comment by mrjoel** [comment]
Great, thanks for the fix!

**Comment by dandavison** [comment]
That's released now (delta 0.8.0).
